source/mainnet/smart-contracts/tutorials/wCCD/wCCD-front-end-set-up.rst

```diff
@@ -4,6 +4,354 @@
+
+    - a running Concordium testnet node
+
+    - have port forwarding enabled with the following command (this step is only required when you run your node on a remote server instead of locally):
+
+    .. note::
+
+        When your node is running on a remote server your cloud provider usually gives you an option to ssh into it.
+        Add the following port forwarding rule to your method to ssh into your instance by running the command in another terminal.
+        The port 10001 on your localhost is forwarded to the port 10001 on your instance.
+
+    .. code-block:: console
+
+        $ssh -NL localhost:10001:<IP-address-of-your-instance>:10001 <username>@<host>
+
+    If you don't have a running testnet node or port forwarding enabled, the piggy bank tutorial :ref:`part 3 <piggy-bank-preparing>`
+    will guide you through these setup steps.
+
+    These prerequisites ensure that you have a testnet node reachable locally on port 10001. The browser wallet requires a
+    `JSON-RPC server <https://github.com/Concordium/concordium-json-rpc>`_
+    that points to your node. Clone the repository with the command:
+
+    .. code-block:: console
+
+        $git clone https://github.com/Concordium/concordium-json-rpc.git
+
+    Build and run the server as described in the README file of the
+    `JSON-RPC repo <https://github.com/Concordium/concordium-json-rpc>`_.
+
+    The final command that you execute to start the ``JSON-RPC server`` is as follows:
+
+    .. code-block:: console
+
+        $yarn start --port 9095 --nodeAddress 127.0.0.1 --nodePort 10001 --nodeTimeout 5000
+
+    Your ``JSON-RPC server`` is running on port `9095` and connects to your local node on port `10001`.
+    You are ready now to build the pre-MVP browser wallet from the source code.
+    Clone the repository with the command:
+
+    .. code-block:: console
+
+        $git clone https://github.com/Concordium/concordium-browser-wallet.git
+
+    You can build the browser wallet extension and load it
+    (``dist`` folder from the path `root/packages/browser-wallet`) into the
+    `Chrome browser <https://developer.chrome.com/docs/extensions/mv3/getstarted/#unpacked>`_ as
+    described in the README file of the
+    `browser wallet repo <https://github.com/Concordium/concordium-browser-wallet/tree/main/packages/browser-wallet>`_.
+
+    .. note::
+
+        Depending on the exact commit hash that you used to build your pre-MVP browser wallet, the
+        screenshots and setup steps might differ. The browser wallet hosts the private keys corresponding
+        to the accounts of the user and a link that points to a ``JSON-RPC server``.
+
+    The next steps are based on the pre-MVP browser wallet from a git commit before around 22.8.2022.
+    These earlier versions of the pre-MVP browser wallet have an input field for the private key
+    to import an already existing account and an input field for the link to the ``JSON-RPC server``.
```

Comment:
The first alternative is not really clear. Perhaps:
`.. which you have to fill out. Alternatively, you can follow..`

Reply:
I've realised that the alternative only relates to the second field, i.e. the rpc server field.

Reply:
It kind of relates to both because the newer browser wallets use seed phrases (and no input of a private key is needed anymore because you can just create a new seed phrase in the wallet)

Reply:
I added this:
 The newer browser wallet from workflow 1 can be used to create new account (no important of private keys needed).